Modules/_ctypes/callproc.c

```diff
@@ -1123,6 +1123,15 @@ PyObject *_ctypes_callproc(PPROC pProc,
                 goto cleanup; /* leaking ? */
             }
         }
+#if (defined(__x86_64__) && (defined(__MINGW64__) || defined(__CYGWIN__))) || \
+        defined(__aarch64__)
+        void *tmp;
+        if (pa->ffi_type->type == FFI_TYPE_STRUCT) {
```

Comment:
Is this definitely the correct test? Similar code to fix bpo-29565 uses a test of the type's size being greater than 8 bytes. Have you tested with structs both under and over this size limit?